lib/OrdinaryDiffEqDifferentiation/src/derivative_utils.jl

```diff
@@ -57,7 +89,29 @@ function calc_tderivative(integrator, cache)
         tf = cache.tf
         tf.u = uprev
         tf.p = p
-        dT = derivative(tf, t, integrator)
+
+        autodiff_alg = alg_autodiff(alg)
+        autodiff_alg = if autodiff_alg isa AutoSparse
+            autodiff_alg = ADTypes.dense_ad(autodiff_alg)
+        else
+            autodiff_alg
+        end
+
+        if alg_autodiff isa AutoFiniteDiff
+            autodiff_alg = SciMLBase.@set autodiff_alg.dir = diffdir(integrator)
+        end
+        
+        if integrator.iter == 1 
+            try 
+                dT = DI.derivative(tf, autodiff_alg, t)
```

Comment:
Is there no preparation for the out-of-place mode? Preparation can still be beneficial in such settings, e.g. with symbolic backends or Mooncake